ipatests/test_integration/test_otp.py

```diff
@@ -309,3 +310,58 @@ def test_2fa_disable_single_prompt(self):
             master.run_command(['ipa', 'user-del', USER2])
             self.master.run_command(['semanage', 'login', '-D'])
             sssd_conf_backup.restore()
+
+    @pytest.fixture
+    def setup_otp_nsslapd(self):
+        # setting nsslapd-idletimeout
+        new_limit = 60
+        conn = self.master.ldap_connect()
+        dn = DN(('cn', 'config'))
+        entry = conn.get_entry(dn)  # pylint: disable=no-member
+        orig_limit = entry.single_value.get('nsslapd-idletimeout')
+        ldap_query = textwrap.dedent("""
+            dn: cn=config
+            changetype: modify
+            replace: nsslapd-idletimeout
+            nsslapd-idletimeout: {limit}
+        """)
+        tasks.ldapmodify_dm(self.master, ldap_query.format(limit=new_limit))
+        # Be sure no services are running and failed units
+        self.master.run_command(['killall', 'ipa-otpd'], raiseonerr=False)
+        check_services = self.master.run_command(
+            ['systemctl', 'list-units', '--state=failed']
+        )
+        assert "0 loaded units listed" in check_services.stdout_text
+        assert "ipa-otpd" not in check_services.stdout_text
+        yield
+        # cleanup
+        tasks.ldapmodify_dm(self.master, ldap_query.format(limit=orig_limit))
+
+    def test_check_otpd_after_dirsrv_restart(self, setup_otp_nsslapd):
+        """Test for OTP when the LDAP connection timed out.
+
+        Test for : https://pagure.io/freeipa/issue/6587
+
+        ipa-otpd was exiting with failure when LDAP connection timed out.
+        Test to verify that when the nsslapd-idletimeout is exceeded (50s idle,
+        60s sleep) then the ipa-otpd process should exit without error.
```

Comment:
You increased the idle to 60s and the timeout to 90s so it doesn't match the comment.

I don't know how much you want to play with this but a 30/45 second idle/sleep would also probably work and would ever so slightly speed up the test.

Reply:
Changed to 30/60 second idle/sleep.
```
Jul 27 15:51:50 master.ipa.test ipa-otpd[23662]: opttestuser@IPA.TEST: response sent: Access-Accept
Jul 27 15:52:50 master.ipa.test systemd[1]: ipa-otpd@1-21071-0.service: Succeeded.
```

Reply:
[logs](http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/e6d75b64-eef0-11eb-bee7-fa163e6f7f4d/)